packages/vm/src/traits.rs

```diff
@@ -0,0 +1,90 @@
+use crate::errors::FfiResult;
```

Comment:
Okay, I cannot think of a better approach, but I have two reactions here:

1. Breaking the equivalence of all traits between wasm and vm doubles our work for testing at least. And doubles the code we need to maintain. Is there a way to avoid breaking this equivalence and somehow have a trait and implementations that work for both the vm and std (changing a bit the function signatures we have in this PR)?
2. If we absolutely must break this equivalence, and can no longer reuse the code, can we gain something more by separating the two interfaces even more? We can add other methods or types to the Vm version, or simplify the wasm version, or such. I don't have a clear idea here, this is more a RFC.

Basically, I would prefer them to either be interoperable, or distinct and well-designed for each use case.

Reply:
I think they should not be interoperable as they fundamentally represent different layers of abstraction in the system, and i have already created a little divergence between them, as Querier now only has one method (the rest were unneeded). As the project evolves, there will probably be even more divergence between them, and i think this decoupling is good for it. 

Maintaining a shared implementation for the two will cause more maintenance overhead than this solution because it's hard (and unnecessary, in my opinion) to accommodate both layers in one definition.

Reply:
Fair enough, then I would say to consider more divergence if it is useful for code correctness, but check with Simon.

I would like to have some simple transforms for the cases where we can (lossily) transform from one api to the other. At least for use in testcases and mocks. (Mentioned above: https://github.com/CosmWasm/cosmwasm/pull/331#discussion_r426479735

Reply:
> but check with Simon

I am super happy to see this split happen. I hit many issues along the way because we reuse types from the standard library for connecting to the backend. I'm sure the implementation can be refined but the direction is the right one in my opinion.

Reply:
Is there anything that is still unresolved here, or can we close this issue?